contracts/modules/cover/CoverViewer.sol

```diff
@@ -2,200 +2,88 @@
-    ProductType memory productType = cover().productTypes(product.productType);
-
-    string memory coverAssetSymbol;
-
-    if (coverData.coverAsset == 0) {
-      coverAssetSymbol = "ETH";
-    } else {
-      address assetAddress = pool().getAsset(coverData.coverAsset).assetAddress;
-      try IERC20Detailed(assetAddress).symbol() returns (string memory v) {
-        coverAssetSymbol = v;
-      } catch {
-        // return coverAssetSymbol as empty string and use coverData.coverAsset instead in the UI
-      }
-    }
-
-    return CoverView(
-      coverData.productId,
-      product.productType,
-      product.yieldTokenAddress,
-      coverData.amountPaidOut,
-      amountRemaining,
-      coverStart,
-      coverEnd,
-      coverData.coverAsset,
-      coverAssetSymbol,
-      productType.claimMethod,
-      gracePeriod
-    );
-  }
-
-  function getPeriods(uint coverId) public view returns (CoverSegment[] memory) {
-    uint segmentCount = cover().coverSegmentsCount(coverId);
-    CoverSegment[] memory segments = new CoverSegment[](segmentCount);
-    for (uint i = 0; i < segmentCount; i++) {
-      segments[i] = cover().coverSegments(coverId, i);
-    }
-    return segments;
-  }
+  function getCovers(uint[] calldata coverIds) external view returns (Cover[] memory) {
+    Cover[] memory covers = new Cover[](coverIds.length);
+    ICover _cover = cover();
 
-  function getCovers(uint[] calldata coverIds) external view returns (CoverView[] memory) {
-    CoverView[] memory coverViews = new CoverView[](coverIds.length);
     for (uint i = 0; i < coverIds.length; i++) {
-      coverViews[i] = getCover(coverIds[i]);
+      uint coverId = coverIds[i];
+
+      CoverData memory coverData = _cover.coverData(coverId);
+      covers[i].productId = coverData.productId;
+      covers[i].coverAsset = coverData.coverAsset;
+      covers[i].amountPaidOut = coverData.amountPaidOut;
+
+      uint segmentsCount = _cover.coverSegmentsCount(coverId);
+      Segment[] memory segments = new Segment[](segmentsCount);
+
+      CoverSegment[] memory coverSegments = _cover.coverSegments(coverId);
+      for (uint segId = 0; segId < segmentsCount; segId++) {
+        CoverSegment memory coverSegment = coverSegments[segId];
+
+        segments[segId].start = coverSegment.start;
+        segments[segId].period = coverSegment.period;
+        segments[segId].gracePeriod = coverSegment.gracePeriod;
+        segments[segId].amount = coverSegment.amount;
+        segments[segId].remainingAmount = coverSegment.amount > coverData.amountPaidOut
+          ? coverSegment.amount - coverData.amountPaidOut
+          : 0;
+      }
+      covers[i].segments = segments;
     }
-    return coverViews;
+    return covers;
   }
 
-  function getCoverSegments(uint coverId) external view returns (CoverSegment[] memory segments) {
-
+  function getCoverSegments(uint coverId) external view returns (Segment[] memory) {
```

Comment:
would be nice if we can unify the code here with the one in `getCovers` at least partially.

Could use a _getCoverSegments that's internal

Reply:
I don't think it's worth it.